src/components/MapView/Analyser/index.tsx

```diff
@@ -0,0 +1,378 @@
+  const boundaryLayerData = useSelector(layerDataSelector(boundaryLayer.id)) as
+    | LayerData<BoundaryLayerProps>
+    | undefined;
+
+  const availableDates = useSelector(availableDatesSelector);
+  const analysisResult = useSelector(analysisResultSelector);
+
+  const isAnalysisLoading = useSelector(isAnalysisLoadingSelector);
+  const isMapLayerActive = useSelector(isAnalysisLayerActiveSelector);
+
+  const [isAnalyserFormOpen, setIsAnalyserFormOpen] = useState(false);
+  const [isTableViewOpen, setIsTableViewOpen] = useState(true);
+
+  // form elements
+  const [hazardLayerId, setHazardLayerId] = useState(
+    (layers.find(layer => layer.type === 'wms') as WMSLayerProps).id,
+  );
+  const [statistic, setStatistic] = useState(AggregationOperations.Mean);
+  const [baselineLayerId, setBaselineLayerId] = useState(
+    (layers.find(layer => layer.type === 'nso') as NSOLayerProps).id,
+  );
+  const [selectedDate, setSelectedDate] = useState<number | null>(null);
+
+  const [belowThreshold, setBelowThreshold] = useState('');
+  const [aboveThreshold, setAboveThreshold] = useState('');
+  const [thresholdError, setThresholdError] = useState<string | null>(null);
+
+  // enforce errors.
+  useEffect(() => {
+    const belowThresholdValue = parseFloat(belowThreshold);
+    const aboveThresholdValue = parseFloat(aboveThreshold);
+    if (belowThresholdValue > aboveThresholdValue) {
+      setThresholdError('Min threshold is larger than Max!');
+    } else setThresholdError(null);
+  }, [belowThreshold, aboveThreshold]);
+
+  // set default date after dates finish loading and when hazard layer changes
+  useEffect(() => {
+    const dates =
+      availableDates[
+        (LayerDefinitions[hazardLayerId] as WMSLayerProps)?.serverLayerName
+      ];
+    if (!dates || dates.length === 0) {
+      setSelectedDate(null);
+    } else setSelectedDate(dates[dates.length - 1]);
+  }, [availableDates, hazardLayerId]);
+
+  const onOptionChange = (setterFunc: (val: any) => void) => (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setterFunc((event.target as HTMLInputElement).value);
+  };
+
+  const adminBoundariesExtent = useMemo(() => {
+    if (!boundaryLayerData)
+      // not loaded yet. Should be loaded in MapView
+      return null;
+    return bbox(boundaryLayerData.data) as Extent; // we get extents of admin boundaries to give to the api.
+  }, [boundaryLayerData]);
+
+  const statisticOptions = map(Object.entries(AggregationOperations), stat => {
+    return (
+      <FormControlLabel
+        key={stat[0]}
+        value={stat[1]}
+        control={<Radio className={classes.radioOptions} size="small" />}
+        label={stat[0]}
+      />
+    );
+  });
+
+  const runAnalyser = async () => {
+    if (!adminBoundariesExtent) return; // hasn't been calculated yet
+    if (analysisResult) {
+      // if one exists we are likely trying to clear it to do a new one
+      dispatch(clearAnalysisResult());
+      return;
+    }
```

Comment:
I'm not sure if I follow...
by moving it into the thunk that would mean clearing in `requestAndStoreAnalysis` if we detect a result already exists?

`Clear Analysis` is its own button and when pressed, we would have to accept `undefined` or data we would never use inside of `requestAndStoreAnalysis` for it to then clear the existing result, making the name ambiguous. Is this what you mean?

Maybe to make things clearer we could move this code from `runAnalysis` to its own function - kept this code in `runAnalysis` for simplicity since we reuse the same button for `Run Analysis` and `Clear Analysis`.

Reply:
Ah, I see - I was just looking at this handler when I made that comment, didn't realize that it was bound to 2 separate buttons.

To me it makes sense to:
1) Have `requestAndStoreAnalysis` clear an existing result iff the result is different from the new one requested (so that the calling function doesn't have to worry/know whether a result exists already).

2) Have a separate, "simpler" clear action for the `Clear Analysis` button.

Does that work?

Reply:
yep, that sounds great. I'll do exactly that :+1: